src/main/cpp/RailroadTile.cpp

```diff
@@ -28,8 +28,13 @@ RailroadTile::getOwner() const
 
 void 
 RailroadTile::setOwner(shared_ptr<Player> o)
-{
+{	
+	if(owner) {
+		owner->setNumRailroads(owner->getNumRailroads() - 1);
+	}
 	owner = o;
```

Comment:
Should this be in a else block? It is overwriting owner right after setting the num railroads, unless I'm missing something.

Reply:
Lines 32-34 decrease the previous owner's railroad count by 1 if there is a previous owner. But if there is no previous owner, then there is no need to run line 33. But, the new owner's railroad count will increase by 1 every time.